bundles/org.openhab.binding.boschshc/src/test/java/org/openhab/binding/boschshc/internal/discovery/ThingDiscoveryServiceTest.java

```diff
@@ -133,73 +122,51 @@ public void testAddDevices() {
-        assertThat(fixture.lastDiscoveryResult.getThingTypeUID(), is(BoschSHCBindingConstants.THING_TYPE_TWINGUARD));
-        assertThat(fixture.lastDiscoveryResult.getThingUID().getId(), is("testDevice_2"));
-        assertThat(fixture.lastDiscoveryResult.getBridgeUID().getId(), is("testSHC"));
-        assertThat(fixture.lastDiscoveryResult.getLabel(), is("Second device"));
-        assertFalse(fixture.lastDiscoveryResult.getProperties().containsKey("Location")); // no rooms given
+        // two calls for the two devices expected
+        verify(discoveryListener, times(2)).thingDiscovered(any(), any());
     }
 
     @Test
     public void testAddDevicesWithNoDevices() {
-        assert fixture != null;
-        assert bridgeHandler != null;
-
-        setMockBridgeAsThingHandler();
-
         ArrayList<Device> emptyDevices = new ArrayList<>();
         ArrayList<Room> emptyRooms = new ArrayList<>();
 
-        assertThat(fixture.thingDiscoveredCalls, is(0));
-        assertNull(fixture.lastDiscoveryResult);
+        verify(discoveryListener, never()).thingDiscovered(any(), any());
 
         fixture.addDevices(emptyDevices, emptyRooms);
 
-        // nothing shall be discovered, but also no exception shall bne thrown
-        assertThat(fixture.thingDiscoveredCalls, is(0));
-        assertNull(fixture.lastDiscoveryResult);
+        // nothing shall be discovered, but also no exception shall be thrown
+        verify(discoveryListener, never()).thingDiscovered(any(), any());
     }
 
     @Test
     public void testAddDevice() {
-        assert fixture != null;
-        assert bridgeHandler != null;
-
-        setMockBridgeAsThingHandler();
-
-        assertThat(fixture.thingDiscoveredCalls, is(0));
-        assertNull(fixture.lastDiscoveryResult);
+        mockBridgeCalls();
 
         Device device = new Device();
         device.deviceModel = "TWINGUARD";
         device.id = "testDevice:ID";
         device.name = "Test Name";
         fixture.addDevice(device, "TestRoom");
 
-        assertThat(fixture.thingDiscoveredCalls, is(1));
-        assertNotNull(fixture.lastDiscoveryResult);
-        assertThat(fixture.lastDiscoveryResult.getBindingId(), is(BoschSHCBindingConstants.BINDING_ID));
-        assertThat(fixture.lastDiscoveryResult.getThingTypeUID(), is(BoschSHCBindingConstants.THING_TYPE_TWINGUARD));
-        assertThat(fixture.lastDiscoveryResult.getThingUID().getId(), is("testDevice_ID"));
-        assertThat(fixture.lastDiscoveryResult.getBridgeUID().getId(), is("testSHC"));
-        assertThat(fixture.lastDiscoveryResult.getLabel(), is("Test Name"));
-        assertThat(fixture.lastDiscoveryResult.getProperties().get("Location").toString(), is("TestRoom"));
+        verify(discoveryListener).thingDiscovered(discoveryServiceCaptor.capture(), discoveryResultCaptor.capture());
+
+        assertThat(discoveryServiceCaptor.getValue().getClass(), is(ThingDiscoveryService.class));
```

Comment:
you could extract `discoveryServiceCaptor.getValue()` to a local variable `discoveryService`